### ISSUE TITLE
Make Vitepress Accessible

### ISSUE THREAD

**Issue by roblevintennis** [vuejs/vitepress#466]
### Is your feature request related to a problem? Please describe.

Yes, when I verify my documentation site at https://agnosticui.com/ with the [IBM Equal Access Accessibility Checker](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp?hl=en-US) it fails hard. This is unfortunate, because one of the things I'm trying to point out is that the UI component library offered is accessible. I lose credibility if the documentation site itself is not a11y compliant.

I actually get even worse results when I scan the vitepress docs themselves. So while this is a feature request, I would also call it a bug. Lighthouse which I would call more permissive also shows errors in the Accessibility category (but if the IBM checker passes Lighthouse will too).

### Describe the solution you'd like

Ensure the code produced by Vitepress is a11y compliant.

I saw violations with the following rules which can easily be cross-referenced at:
https://www.ibm.com/able/requirements/requirements

1.3.1 Info and Relationships
1.4.3 Contrast (Minimum)
2.4.1 Bypass Blocks
2.5.3 Label in Name
4.1.2 Name, Role, Value

_Note: this is only from scanning the top-level page. I saw a couple of other warnings on other pages, but this is a good start._

### Describe alternatives you've considered

There are no alternatives. These issues are making Vitepress generated sites less inclusive than they could be meaning certain site users are being excluded which is a real bummer since Vitepress is otherwise producing such lovely sites.

### Additional context

![image](https://user-images.githubusercontent.com/142403/145711147-8be2df36-576a-4ba4-9600-b08fae40e1d5.png)

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/coc)
- [X] Read the [docs](https://vitepress.vuejs.org/).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/master/.github/contributing.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by kiaking** [comment]
Can someone run this again with the latest version and see how we're doing? 👀

**Comment by kiaking** [comment]
Closing due to inactivity. We would like to make VitePress accessible, but this issue is too huge it's not workable. If we have an issue, let's create smaller issues where we can take action on it.
